### PR TITLE
Material buttons - before in #1090

### PR DESCRIPTION
**Pull request by davivel**
- [x] BUG: https://github.com/owncloud/android/issues/1381 : fixed

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
This is the successor of #1090 since I accidentally pushed it to master but David fixed that for me and we ended up having to create a new PR.

Can someone give this PR some review love? @tobiasKaminsky @LukeOwncloud maybe?

The code changes are mostly theming, layout files and the images for material design check boxes :disappointed: 

@LukeOwncloud I added you simple since you seem to be quite active here and besides @tobiasKaminsky might be able to do a community code review so we can take this of the dev team' shoulders :smiley:

**Comment by AndyScherzinger** [comment]
@jancborchardt changed the color of the divider according to the discussion on the beta branch for the material buttons branch https://github.com/owncloud/android/commit/2afd2d6fb6171e94fd40a712cab9ee036b4c8058

**Comment by AndyScherzinger** [comment]
#1232 is fixed by this PR.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can you do a code review on this PR? :smiley: very small one!

**Comment by jancborchardt** [comment]
Mabe @LukeOwncloud @Kernald @Stoyicker can review this as well. :)

**Comment by tobiasKaminsky** [comment]
:+1: looks good to me.
And as it is already included in beta, it can be tested out in the wildness ;)

**Comment by LukasReschke** [comment]
The history looks a little bit often merged… http://kernowsoul.com/blog/2012/06/20/4-ways-to-avoid-merge-commits-in-git/

screen shot 2015-11-04 at 18 03 59

**Comment by AndyScherzinger** [comment]
Uh, nice! Never knew how to not have the merge Commits. Thanks for the hint, will incorporate this config in the future! :+1:

**Comment by LukeOwncloud** [comment]
I did not review the code, but I tested the new design using the beta version. Looks good, works fine, except for this tiny comment: https://github.com/owncloud/android/pull/1100#issuecomment-154081397

**Comment by AndyScherzinger** [comment]
@LukeOwncloud the fab isn't part of this PR ;) - As for the rebase thing, I will need some more experience with that since obviously pull and rebase are quite different and while the pull works fine (I pulled from several branches) the rebase simply gives me conflicts... So staying with pull for now simply for speed reasons ;)

**Comment by jancborchardt** [comment]
It seems like this PR gets dangerously big (difficilt to review) and we should really prioritize it in reviewing. cc @MTRichards @davivel @masensio

**Comment by AndyScherzinger** [comment]
@jancborchardt this PR is dead simple to review if you just look at the changes (in the other tab), the long history in this PR basically is due to the number of merges of the changes done on the master branch

**Comment by jancborchardt** [comment]
True, looking at the code. :D I now saw it linked from so many places that I was under the impression it was full of different things.

**Comment by AndyScherzinger** [comment]
It's is a fairly minor change with could then be utilized in many places looking at the latest added features. This PR is rather an enabler for others :)

**Comment by tobiasKaminsky** [comment]
As it is completly a design question, maybe it is a good idea to show screenshots of the current state of this PR for @jancborchardt to get his ok.
Then the code review should be very fast?

**Comment by AndyScherzinger** [comment]
I already have the OK from @jancborchardt (https://github.com/owncloud/android/pull/1090#issuecomment-147176259) and the screenshots are in the original PR #1090 :)

**Comment by tobiasKaminsky** [comment]
Ahh :/
This is so hard to keep this all in my mind ;)
Sorry for this.

**Comment by AndyScherzinger** [comment]
No worries! :) But that is why I keep bugging everyone to get back to merging community PRs as soon as the core team finds the time since I can't keep all this in my head either like running several feature branches with PRs in parallel. Thus it will get to complex at some point, renders PRs unmergeable or blocks further community developments (except when we abandon PRs, which I am not really willing to)

**Comment by AndyScherzinger** [comment]
@jancborchardt @davivel @masensio @rperezb just pinging you again about this PR since as I mentioned it before is a fairly small change (ignoring all the master merges), centralized some design values, adds proper material checkboxes and gives you a style for buttons to display them as primary actions :)

Plus: There are no merge conflicts :grinning:

**Comment by jancborchardt** [comment]
Well, I’m totally down with this, just not much I can do in the way of testing. :)

**Comment by davivel** [comment]
@AndyScherzinger , @jancborchardt , I'm resuming work here.

File: res/layout-v14/generic_explanation.xml [hunk 1]
```diff
@@ -49,15 +49,17 @@
         android:orientation="horizontal" >
 
         <!-- 'OK' / 'CANCEL' BUTTONS CHANGE THEIR ORDER FROM ANDROID 4.0 ; THANKS, GOOGLE -->
-        <Button
+        <android.support.v7.widget.AppCompatButton
```
Comment (davivel) [inline res/layout-v14/generic_explanation.xml]: Now v14 is the minimum valid API; there is no need to have two separece generic_explanation.xml files. This one should replace the one in res/layout, then the folder layout-v14 may be removed.

File: res/layout/drawer.xml [hunk 2]
```diff
@@ -62,22 +62,22 @@
...
             android:layout_height="100dp"
-            android:paddingTop="16dp"
-            android:paddingBottom="16dp"
+            android:paddingTop="@dimen/standard_padding"
+            android:paddingBottom="@dimen/standard_padding"
             android:background="@color/owncloud_blue_accent">
 
             <ImageView
                 android:id="@+id/itemIcon"
                 android:layout_width="24sp"
```
Comment (davivel) [inline res/layout/drawer.xml]: Mmm, wouldn't be better ```24dp```? AFAIK ```sp``` units only make sense for text.

File: res/layout/log_send_file.xml [hunk 3]
```diff
@@ -49,11 +49,14 @@
...
         android:layout_width="fill_parent"
         android:layout_height="wrap_content"
-        android:orientation="horizontal" >
+        android:orientation="horizontal"
+        android:layout_marginBottom="@dimen/standard_margin"
+        android:layout_marginLeft="@dimen/standard_margin"
+        android:layout_marginRight="@dimen/standard_margin">
 
-    <android.support.v7.widget.AppCompatButton
+    <Button
```
Comment (davivel) [inline res/layout/log_send_file.xml]: Is this intended?

File: src/com/owncloud/android/ui/activity/FileActivity.java [hunk 4]
```diff
@@ -443,6 +438,21 @@ public void onDrawerOpened(View drawerView) {
...
 
     /**
+     * sets the given account name in the drawer in case the drawer is available. The account name
+     * is shortened beginning from the @-sign in the username.
+     *
+     * @param navigationDrawerLayout the drawer layout to be used
+     * @param account                the account to be set in the drawer
+     */
+    protected void setUsernameInDrawer(RelativeLayout navigationDrawerLayout, Account account) {
+        if (navigationDrawerLayout != null && getAccount() != null) {
```
Comment (davivel) [inline src/com/owncloud/android/ui/activity/FileActivity.java]: Shouldn't be ```&& account != null)``` ?

Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/activity/FileActivity.java]: Yes you're right!

File: res/layout/log_send_file.xml (hunk 3 above)
Comment (AndyScherzinger) [inline res/layout/log_send_file.xml]: True, it should be AppCompatButton

File: res/layout/drawer.xml (hunk 2 above)
Comment (AndyScherzinger) [inline res/layout/drawer.xml]: Yes should be dp

File: res/layout-v14/generic_explanation.xml (hunk 1 above)
Comment (AndyScherzinger) [inline res/layout-v14/generic_explanation.xml]: True, I'll do the house keeping and remove the v14 structure

**Comment by davivel** [comment]
@AndyScherzinger , did you start coding the changes? I did it locally to start testing. Sorry, I should have warned you before :$ ...

**Comment by AndyScherzinger** [comment]
No I didn't, so feel free to push :)

**Comment by davivel** [comment]
Code review is :+1: .

**Comment by AndyScherzinger** [comment]
Thanks for your efforts @davivel :+1:

**Comment by davivel** [comment]
Scheduled for next release :)

**Comment by AndyScherzinger** [comment]
Wow nice! Exspecially seeing the process behind QA here. Impressive! :+1:

**Comment by AndyScherzinger** [comment]
@davivel I merged the master branch changes and resolved the conflicts surrounding the color changes for the setup.xml you did. Please have a look if it's fine. I did a check and ran the branch on my phone and everything looks fine to me. :smiley:

**Comment by jancborchardt** [comment]
Oooooooh nice! :) Finally merged.

**Pull request merged**